iradix.go

```diff
@@ -183,6 +183,28 @@ func (t *Txn) writeNode(n *Node, forLeafUpdate bool) *Node {
 	return nc
 }
 
+// Visit all the nodes in the tree under n, and add their mutateChannels to the transaction
+// Returns the size of the subtree visited
+func (t *Txn) recursiveTrackChannelAndCount(n *Node) int {
```

Comment:
Much simpler method that both counts the nodes visited and tracks mutate channels correctly